tests/test_opnieuw.py

```diff
@@ -65,9 +66,44 @@ def test_retry_times(self):
         except TypeError as e:
             end = time.time()
             t_diff = end - start
-            self.assertGreater(t_diff, 1)
+            self.assertGreater(t_diff, 0.5)
             self.assertEqual(self.counter, 3)
 
+    def test_retry_immediately_global(self) -> None:
+        start = time.time()
```

Comment:
`time.time()` is not suitable for measuring durations, because it is subject to various corrections. For example, it can jump backwards a second on leap seconds, but the more common case is the NTP daemon adjusting the system clock when it has drifted out of sync, for example after waking from hibernate, or when resuming a virtual machine that was suspended for a long time. With `time.time()`, you may even find `t_diff` to be negative!

The right tool for measuring elapsed time is [`time.monotonic()`](https://docs.python.org/3/library/time.html?highlight=time%20time#time.monotonic). The monotonic clock does not go backwards, so `t_diff` will not be negative.

Reply:
Ahh cool, thanks for clearing that up. `time.monotonic` it is.